tracing/tracing/src/main/java/io/helidon/tracing/SpanLifeCycleListener.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2024 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.tracing;
+
+/**
+ * Behavior of listeners notified of span life cycle events.
+ * <p>
+ * The interface declares default empty implementations for each method, allowing concrete implementations of the
+ * interface to implement only the methods that are relevant for them.
+ * <p>
+ * Helidon invokes the applicable methods of a life cycle listener in the following order:
+ * <table>
+ *     <caption style="text-align:left">Order of Invocation of Listener Methods</caption>
+ *
+ *     <tbody>
+ *         <tr>
+ *         <th style="text-align:left;width:15%">Method</th>
+ *         <th style="text-align:left">When invoked</th>
+ *     </tr>
+ *     <tr>
+ *         <td>{@linkplain #beforeStart(io.helidon.tracing.Span.Builder) beforeStart}</td>
```

Comment:
I wonder if this event could conceptually be renamed to "building". Then you would have `building`, `started`, etc. and it might be more consistent.

Reply:
I like the idea, though I feel like I'm leaning toward "starting." It's more specific to spans than "building."

Reply:
Changed